spec.md

```diff
@@ -336,17 +336,44 @@ on the definition of OPTIONAL.
   event format defines the relationship in
   [section 3.1](./json-format.md#31-special-handling-of-the-data-attribute).
 
-  When this attribute is omitted, the `data` attribute simply follows the event
-  format's encoding rules. For the JSON event format, the `data` attribute value
-  can therefore be a JSON object, array, or value.
-
   For the binary mode of some of the CloudEvents transport bindings, where the
   `data` content is immediately mapped into the payload of the transport frame,
   this field is directly mapped to the respective transport or application
   protocol's content-type metadata property. Normative rules for the binary mode
   and the content-type metadata mapping can be found in the respective transport
   mapping specifications.
 
+  The attribute MUST be set for Binary content in the data field, with an 
+  accurate media-type. If no IANA registered media-type is available, a vendor 
+  extension or application/octet-stream MAY be used.
+
+  The attribute SHOULD be set for String content in the data field. If set, the
```

Comment:
The CloudEvents-String is defined as "Sequence of **printable** Unicode characters". That (probably) makes sense for a regular attribute, but for `data`?

If a publisher wants to send text-data that (potentially) contains whitespaces, they can use `Binary`, but will still declare `text/plan`. How can a consumer decide if it's a `String` or a `Binary`?

Reply:
Correcting myself: The publisher doesn't want to use `Binary`, because then they have to `base64` encode.

I guess it is out of scope for this PR, but I suggest we split `String` into two parts: The `String` we need as an identifier for the `Map`, and the `String` we want to use for attributes. The latter should not be constraint to printable characters IMO. We could also further constrain the `Map-String` type (e.g. SHOULD follow the same pattern for attribute names).